types/module/module.go

```diff
@@ -17,49 +17,54 @@ process. This separation is necessary, however we still want to allow for a
 high level pattern for modules to follow - for instance, such that we don't
 have to manually register all of the codecs for all the modules. This basic
 procedure as well as other basic patterns are handled through the use of
-ModuleBasicManager.
+BasicManager.
 */
-package types
+package module
 
 import (
 	"encoding/json"
 
-	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/gorilla/mux"
 	"github.com/spf13/cobra"
 	abci "github.com/tendermint/tendermint/abci/types"
-)
 
-// ModuleClient helps modules provide a standard interface for exporting client functionality
-type ModuleClient interface {
-	GetQueryCmd() *cobra.Command
-	GetTxCmd() *cobra.Command
-}
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
 
 //__________________________________________________________________________________________
 // AppModuleBasic is the standard form for basic non-dependant elements of an application module.
 type AppModuleBasic interface {
 	Name() string
 	RegisterCodec(*codec.Codec)
+
+	// genesis
 	DefaultGenesis() json.RawMessage
 	ValidateGenesis(json.RawMessage) error
+
+	// client functionality
+	RegisterRESTRoutes(context.CLIContext, *mux.Router, *codec.Codec)
+	GetTxCmd() *cobra.Command
+	GetQueryCmd() *cobra.Command
 }
 
 // collections of AppModuleBasic
-type ModuleBasicManager []AppModuleBasic
+type BasicManager []AppModuleBasic
 
-func NewModuleBasicManager(modules ...AppModuleBasic) ModuleBasicManager {
+func NewBasicManager(modules ...AppModuleBasic) BasicManager {
 	return modules
 }
 
 // RegisterCodecs registers all module codecs
-func (mbm ModuleBasicManager) RegisterCodec(cdc *codec.Codec) {
+func (mbm BasicManager) RegisterCodec(cdc *codec.Codec) {
```

Comment:
now that `ModuleBasicManager` got renamed `BasicManager` should `mbm` be renamed `bm` ?

Reply:
sure - updated